internal/driver/glfw/file.go

```diff
@@ -6,48 +6,49 @@ import (
 	"path/filepath"
 
 	"fyne.io/fyne"
+	"fyne.io/fyne/storage"
 )
 
 type file struct {
 	*os.File
 	path string
 }
 
-func (d *gLDriver) FileReaderForURI(uri string) (fyne.FileReadCloser, error) {
+func (d *gLDriver) FileReaderForURI(uri fyne.URI) (fyne.FileReadCloser, error) {
 	return openFile(uri, false)
 }
 
 func (f *file) Name() string {
 	return filepath.Base(f.path)
 }
 
-func (f *file) URI() string {
-	return "file://" + f.path
+func (f *file) URI() fyne.URI {
+	return storage.NewURI("file://" + f.path)
 }
 
 type fileWriter struct {
 	*os.File
 	path string
 }
 
-func (d *gLDriver) FileWriterForURI(uri string) (fyne.FileWriteCloser, error) {
+func (d *gLDriver) FileWriterForURI(uri fyne.URI) (fyne.FileWriteCloser, error) {
 	return openFile(uri, true)
 }
 
 func (f *fileWriter) Name() string {
 	return filepath.Base(f.path)
 }
 
-func (f *fileWriter) URI() string {
-	return "file://" + f.path
+func (f *fileWriter) URI() fyne.URI {
+	return storage.NewURI("file://" + f.path)
 }
 
-func openFile(uri string, create bool) (*file, error) {
-	if len(uri) < 8 || uri[:7] != "file://" {
+func openFile(uri fyne.URI, create bool) (*file, error) {
+	if uri.Scheme() != "file" {
 		return nil, fmt.Errorf("invalid URI for file: %s", uri)
 	}
 
-	path := uri[7:]
+	path := uri.String()[7:]
```

Comment:
I agree adding "func Path() string" to the interface would be confusing, but perhaps we can come up with something else to avoid string slicing all together?

Reply:
I'm not really concerned as this is just in driver code, but we should revisit when support for remote files is added later